src/pages/assetGroup/AssetGroup.jsx

```diff
@@ -20,14 +27,59 @@ import MoreMenu from '../../components/MoreMenu';
 import ConfirmDelete from '../../components/ConfirmDelete';
 import EntityHeader from '../../components/EntityHeader';
 import AGSTable from './AGSTable';
+import AssetsProcessingAlert from '../../components/assets/AssetsProcessingAlert';
+import { defaultCrossfadeDuration } from '../../constants/defaults';
+import { progressState } from '../../constants/progress';
+
+const MAX_BUFFER_TIME = 8000; // milliseconds
 
 export default function AssetGroup() {
-  const { id } = useParams();
+  const mountTime = useMemo(() => Date.now(), []);
+  const { id: guid } = useParams();
   const history = useHistory();
   const queryClient = useQueryClient();
   const intl = useIntl();
 
-  const { data, loading, error, statusCode } = useAssetGroup(id);
+  const { data, isLoading, error } = useAssetGroupQuery(guid, {
+    notifyOnPropsChange: ['data', 'isLoading', 'error'],
+  });
+
+  const assetGroupSightings = get(data, 'asset_group_sightings', []);
+
+  const isAssetGroupPreparing =
+    assetGroupSightings[0]?.stage === agsStage.preparation;
+
+  const preparationProgressGuid = get(
+    data,
+    'progress_preparation.guid',
+  );
+
+  const isProgressQueryEnabled = Boolean(
+    preparationProgressGuid && isAssetGroupPreparing,
+  );
+
+  const {
+    state: preparationProgressState,
+    error: preparationProgressError,
+  } = useProgressStateQuery(
```

Comment:
i'm really confused about the role this hook is playing. why not just `useProgressQuery` here and parse `data` with the utilities you have made to determine the status code and whether it's in a resolved state

Reply:
If progress is pending, `useProgressQuery` will cause everything to re-render every two seconds. This hook triggers re-renders far less often. For AssetGroup, this may be overkill, I'm not sure, but for the sightings page, it will probably become more relevant.

Reply:
i think it might be premature optimization.. while we want to avoid unnecessary rerenders usually what we are worried about is something causing a ton of rerenders per second, one render every two seconds should be no big deal unless the component being rendered has a ton of issues